backend/compact-connect/lambdas/cloudfront-csp/index.js

```diff
@@ -1,5 +1,337 @@
+ * @return {string}             The prepped src list string;
+ */
+const buildSrcString = (name = '', list = []) => {
+    let src = [];
+    let srcString = '';
+
+    if (Array.isArray(list)) {
+        src = src.concat(list);
+    }
+
+    srcKeywordsEscape(src, name);
+
+    if (src.length) {
+        srcString = `${name} ${src.join(' ')};`;
+    }
+
+    return srcString;
+};
+
+// ============================================================================
+//                               RESPONSE HEADERS                             =
+// ============================================================================
+/**
+ * Set the CSP header on the response (by reference).
+ * @param {string} requestDomain The domain making the request.
+ * @param {object} [headers={}]  The event response headers (updated by reference).
+ */
+const setCspHeader = (requestDomain, headers = {}) => {
+    const domains = getEnvironmentUrls(requestDomain);
+
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
+    headers['content-security-policy'] = [{
+        key: 'Content-Security-Policy',
+        value: `${[
+            `default-src 'none';`,
+            buildSrcString('manifest-src', [
+                'self',
+            ]),
+            buildSrcString('script-src', [
+                'self',
+            ]),
+            buildSrcString('script-src-elem', [
+                'self',
+            ]),
+            buildSrcString('script-src-attr', [
+                'self',
+            ]),
+            buildSrcString('worker-src', [
+                'self',
+            ]),
+            buildSrcString('style-src', [
+                'self',
+                'https://fonts.googleapis.com',
+            ]),
+            buildSrcString('style-src-elem', [
+                'self',
+                'https://fonts.googleapis.com',
+            ]),
+            buildSrcString('style-src-attr', [
+                'self',
+            ]),
+            buildSrcString('font-src', [
+                'self',
+                'https://fonts.gstatic.com',
+            ]),
+            buildSrcString('img-src', [
+                'self',
+                'data:',
+                domains.dataApi,
+            ]),
+            buildSrcString('media-src', [
+                'self',
+                domains.dataApi,
+            ]),
+            buildSrcString('frame-src', [
+                'self',
+            ]),
+            buildSrcString('frame-ancestors', [
+                'none'
+            ]),
+            buildSrcString('object-src', [
+                'none',
+            ]),
+            buildSrcString('form-action', [
+                'none',
+            ]),
+            buildSrcString('connect-src', [
+                'self',
+                domains.dataApi,
+                domains.s3Upload,
+                'https://cognito-idp.us-east-1.amazonaws.com',
```

Comment:
can we hoist this URL up like we did in the test suite?

Reply:
Done.